src/Vegetation/PlantHydraulics.jl

```diff
@@ -260,34 +258,112 @@ to take into account the change in K_sat halfway
 between z1 and z2. The expression is given in full in the Clima docs
 in section 6.4 "Bulk plant hydraulics model".
 """
-function flux(z1, z2, ψ1, ψ2, vg_α, vg_n, vg_m, ν, S_s, K_sat1, K_sat2)
+function flux(z1, z2, ψ1, ψ2, K1, K2)
+    
+    flux = -(K1 + K2) / 2 * ((ψ2 - ψ1) / (z2 - z1) + 1)
+    
+    return flux # (m/s)
+end
 
-    S_l1 = inverse_water_retention_curve(vg_α, vg_n, vg_m, ψ1, ν, S_s)
-    S_l2 = inverse_water_retention_curve(vg_α, vg_n, vg_m, ψ2, ν, S_s)
+"""
+    AbstractConductivityModel{FT <: AbstractFloat}
 
-    K1_ψ1 = hydraulic_conductivity(K_sat1, vg_m, S_l1)
-    K2_ψ2 = hydraulic_conductivity(K_sat2, vg_m, S_l2)
+An abstract type for the plant hydraulics conductivity model.
+"""
+abstract type AbstractConductivityModel{FT <: AbstractFloat} end
 
-    flux = -(K1_ψ1 + K2_ψ2) / 2 * ((ψ2 - ψ1) / (z2 - z1) + 1)
+"""
+    AbstractRetentionModel{FT <: AbstractFloat}
 
-    return flux # (m/s)
-end
+An abstract type for the plant hydraulics conductivity model.
+"""
+abstract type AbstractRetentionModel{FT <:  AbstractFloat} end
 
 """
-     hydraulic_conductivity(K_sat::FT, vg_m::FT, S::FT) where {FT}
+    Weibull{FT} <: AbstractConductivityModel{FT}
+The necessary parameters for the Weibull plant hydraulic conductivity model. 
+Default parameters were taken from Natan Holtzman's orginal Ozark test.
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct Weibull{FT, T1, T2} <: AbstractConductivityModel{FT}
+    "The absolute water pressure in xylem (or xylem water potential) at which 50% 
+    of maximum xylem conductance is lost (Liu, 2020). It shifts the curve along 
+    the pressure axis (Sperry, 2016)."
+    ψ50::T1
+    "Parameter c controls shape (‘exponential’ shape with no threshold has c ≤ 1, sigmoidal
+    threshold has c > 1 (Sperry, 2016)."
+    c::T2
+end
 
+"""
+    hydraulic_conductivity(conductivity_params::Weibull{FT}, ψ::FT) where {FT}
 A point-wise function returning the hydraulic conductivity, using the
-van Genuchten formulation.
+Weibull formulation.
 """
-function hydraulic_conductivity(K_sat::FT, vg_m::FT, S::FT) where {FT}
-    if S < FT(1)
-        K = sqrt(S) * (FT(1) - (FT(1) - S^(FT(1) / vg_m))^vg_m)^FT(2)
+function hydraulic_conductivity(conductivity_params::Weibull{FT}, ψ::FT) where {FT}
+    (; K_sat, ψ50, c) = conductivity_params
+    if ψ <= FT(0) 
+        K = exp(-(ψ / ψ50)^c) 
     else
         K = FT(1)
     end
     return K * K_sat # (m/s)
 end
 
+"""
+    linear{FT} <: AbstractRetentionModel{FT}
```

Comment:
maybe linear_retention_curve?

is intercept such that when S = 1, psi = 0?

Reply:
Hi Katherine! I fixed this. Btw I found an excellent paper on water potential in wood. The best (most rigorous) I have found so far. (A. Cloutier; Y. Fortin, 1991)